### DISCUSSION TITLE
Detect deploy hook build on Vercel?

### DISCUSSION THREAD

**Discussion by nlswtlr** [vercel/community#740] [Help] [answered]
Hi, 

is there a way I can detect if a deploy hook was triggering the deploy?
In particular I am thinking about making use of a bash or Node script in "Ignored Build Step".

**Use case:**
For all non production branches I want to use the default behaviour (push & deploy). For the production branch though all the tests should pass (using GitLab runners) and just then I want to trigger the deployment.

**Comment by btbenedi** [comment, marked as answer]
I have another use case for this:

I have a monorepo multi-zone setup, whenever files in a zone directory change, we do the build as normal, I have the ```Ignored Build Step``` functioning for this, so if no files change in ```/<dir_a>/```, the build gets skipped, however if a deploy hook was fired for that project, it skips because no files changed, I want the ```Ignored Build Step``` to return 1 if it's a deploy hook triggering the build as the web hooks are fired by the CMS, which means we have new content and need to rebuild.

**Reply by amyegan** [reply]
Hi @btbenedi. Another community member shared a solution that lets them bypass their other build restrictions by detecting deploy hook in their Ignored Build Step script https://github.com/vercel/community/discussions/285#discussioncomment-1696833

**Reply by btbenedi** [reply]
thank you @amyegan ! I'll take a look through this thread
